server.go

```diff
@@ -324,190 +324,64 @@ func (s *Server) handlerAddMovie(w http.ResponseWriter, r *http.Request) {
+			s.l.Debug("autofill")
+			results, links := s.handleAutofill(&data, w, r)
 
-			if len(errors) > 0 {
-				errText = append(errText, errors...)
+			if results == nil || links == nil {
+				data.ErrorMessage = append(data.ErrorMessage, "Could not autofill all fields")
 				data.ErrAutofill = true
-
-				if rerenderSite {
-					data.ErrorMessage = errText
-					if err := s.executeTemplate(w, "addmovie", data); err != nil {
-						s.l.Error("Error rendering template: %v", err)
-					}
-					return
-				}
 			} else {
+				// Fill all the fields in the movie struct
 				movie.Name = results[0]
 				movie.Description = results[1]
 				movie.Poster = filepath.Base(results[2])
+				movie.Remarks = results[3]
 				movie.Links = links
-				movie.Remarks = remarks
-			}
-		} else {
-			s.l.Debug("Autofill not checked")
-
-			// Here we continue with the other input checks
-			maxTitleLength, err := s.data.GetCfgInt(ConfigMaxTitleLength, DefaultMaxTitleLength)
-			if err != nil {
-				s.l.Error("Unable to get %q: %v", ConfigMaxTitleLength, err)
-				s.doError(
-					http.StatusInternalServerError,
-					"something went wrong :C",
-					w, r)
-				return
-			}
+				movie.AddedBy = user
 
-			data.ValTitle = strings.TrimSpace(r.FormValue("MovieName"))
-			if len(data.ValTitle) > maxTitleLength {
-				s.l.Debug("Title too long: %d", len(data.ValTitle))
-				data.ErrTitle = true
-				errText = append(errText, "Title too long!")
-			} else if len(data.ValTitle) == 0 {
-				s.l.Debug("Title too short: %d", len(common.CleanMovieName(data.ValTitle)))
-				data.ErrTitle = true
-				errText = append(errText, "Title too short!")
-			}
+				// Prepare a int for the id
+				var movieId int
 
-			movieExists, err := s.data.CheckMovieExists(r.FormValue("MovieName"))
-			if err != nil {
-				s.doError(
-					http.StatusInternalServerError,
-					fmt.Sprintf("Unable to check if movie exists: %v", err),
-					w, r)
-				return
-			}
-
-			if movieExists {
-				data.ErrTitle = true
-				s.l.Debug("Movie exists")
-				errText = append(errText, "Movie already exists")
-			}
-
-			if data.ValTitle == "" && !(r.FormValue("AutofillBox") == "on") {
-				errText = append(errText, "Missing movie title")
-				data.ErrTitle = true
-			}
-
-			descr := strings.TrimSpace(r.FormValue("Description"))
-			data.ValDescription = descr
-
-			maxDescriptionLength, err := s.data.GetCfgInt(ConfigMaxDescriptionLength, DefaultMaxDescriptionLength)
-			if err != nil {
-				s.l.Error("Unable to get %q: %v", ConfigMaxTitleLength, err)
-				s.doError(
-					http.StatusInternalServerError,
-					"something went wrong :C",
-					w, r)
-				return
-			}
-
-			if len(data.ValDescription) > maxDescriptionLength {
-				s.l.Debug("Description too long: %d", len(data.ValDescription))
-				data.ErrDescription = true
-				errText = append(errText, "Description too long!")
-			}
+				movieId, err = s.data.AddMovie(movie)
+				if err != nil {
+					s.l.Error("Movie could not be added. Error: %v", err)
```

Comment:
A generic error should probably be displayed to the user here (by adding to the `data.ErrorMessage` slice).

Reply:
Will be changed in the next commit